docs/sts-migrate-to-private-bucket.md

```diff
@@ -0,0 +1,428 @@
+# AWS Security Token Service - Steps to Migrate the OIDC issuer from public S3 Bucket to CloudFront Distribution
+
+> NOTE: the steps described in this document are not supported or recommended to be done in a production environment.
+
+- [Prerequisites](#Prerequisites)
+- [Test the existing token](#test-token)
+- [Setup new OIDC with CloudFront Distribution](#setup)
+    - [Create the CloudFront Distribution](#setup-cloudfront)
+    - [Create and patch the new OIDC discovery documents and JWKS](#setup-oidc-documents)
+    - [Create the new OIDC using CloudFront Distribution](#setup-oidc-idp)
+- [Backing up current cluster data](#backup)
+- [Patch the cluster to use the new OIDC](#patch-cluster)
+- [Revoke public access to S3 Bucket](#revoke-s3-public-access)
+- [Rollback to OIDC with S3 Public URL](#rollback)
+- [Rollback to OIDC with S3 Public URL](#delete)
+
+## Prerequisites <a name="Prerequisites"></a>
+
+- An OCP Cluster created on AWS with manual authentication mode with STS using S3 Bucket as OIDC URL (issuerURL)
+
+
+- Environment variables exported:
+```bash
+export CLUSTER_NAME="<ChangeMe>"
+export OIDC_BUCKET_HOST=$(oc get authentication cluster -o jsonpath={'.spec.serviceAccountIssuer'} | awk -F'://' '{print$2}')
+export OIDC_BUCKET_NAME=$(oc get authentication cluster -o jsonpath={'.spec.serviceAccountIssuer'} | awk -F'://' '{print$2}' |awk -F'.' '{print$1}')
+export CLUSTER_REGION=$(oc get authentication cluster -o jsonpath={'.spec.serviceAccountIssuer'} | awk -F'://' '{print$2}' |awk -F'.' '{print$3}')
+```
+
+- Make sure you can reach (read) the S3 Bucket created with the default name
+
+```bash
+aws s3 ls --region $CLUSTER_REGION s3://$OIDC_BUCKET_NAME
+```
+
+- A clean work directory: a lot of files will be created, make sure you switched to a new work directory to save the files properly (it can be used in the future for rollback)
+
+## Test the existing token <a name="test-token"></a>
+
+This section is to make sure everything is working correctly in your existing environment.
+
+The steps described below will use the credentials provided to the `machine-api` component, trying to assume the role using `aws-cli`.
+
+It's expected that the existing token will be able to authenticate in AWS. Otherwise, you must abort the operations and do not try to run any step described in the next sections
+
+```bash
+## test existing token
+# Get Token path from AWS credentials mounted to pod
+TOKEN_PATH=$(oc get secrets aws-cloud-credentials \
+    -n openshift-machine-api \
+    -o jsonpath='{.data.credentials}' |\
+    base64 -d |\
+    grep ^web_identity_token_file |\
```

Comment:
```suggestion
    grep "^web_identity_token_file" |\
```

I ran into an issue re-running this and quoting the grep solved the problem for me. Local fedora grep difference maybe?

```sh
# Copy pasting into terminal
➜  TOKEN_PATH=$(oc get secrets aws-cloud-credentials \                                                                                                                  cmdsubst>     -n openshift-machine-api \
cmdsubst>     -o jsonpath='{.data.credentials}' |\
cmdsubst pipe>     base64 -d |\
cmdsubst pipe pipe>     grep ^web_identity_token_file |\
cmdsubst pipe pipe pipe>     awk '{print$3}')
grep: ccoctl: binary file matches
grep: current-cluster: Is a directory
grep: new-oidc: Is a directory
```